### DISCUSSION TITLE
tutorials/voyage-part2/04-auth-in-a-supergraph

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#90] [course feedback]
# tutorials/voyage-part2/04-auth-in-a-supergraph

https://www.apollographql.com/tutorials/voyage-part2/04-auth-in-a-supergraph

**Comment by torressam333** [comment]
Is it typical for all subgraphs to use the same auth mechanism? What if each subgraph implemented it's own authentication scheme, would we configure multiple tokens in the config.yaml per subgraph/service?

**Reply by slongato** [reply]
This is a late response, but if anyone else is also curious, Apollo has some docs on using different secrets for `Router-Authorization' [here](https://www.apollographql.com/docs/graphos/graphs/securing-a-subgraph/#step-2-add-the-secret-to-your-router-config/) as well as different [http header rules here](https://www.apollographql.com/docs/graphos/cloud-routing/configuration/#http-header-rules).

**Reply by mabuyo** [reply]
Aaah, we definitely missed this comment; thank you for the helpful response @slongato!
